### DISCUSSION TITLE
New datasets and front page for the first release

### DISCUSSION THREAD

**Discussion by jeanfeydy** [kernel-matrix-benchmarks/kernel-matrix-benchmarks#3] [General]
Hi @maltezfaria, @PierreMarchand20,

As discussed this morning, you should now have push permission on the repo.

The important files in the repo are highlighted [here](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks#main-files).
In order to add a new dataset, you may look at:

- [datasets.py](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/kernel_matrix_benchmarks/datasets.py).
- [create_website_local.sh](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/create_website_local.sh).
- [create_website_AWS.sh](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/create_website_AWS.sh).

Moreover, if you want to work on improving the visualization for collections of "datasets" (e.g. to put time vs npoints figures on the front page), you may have a look at:

- [create_website.py](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/create_website.py).
- [The Jinja template for the front page](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/templates/summary.html).
- [plot.py](https://github.com/kernel-matrix-benchmarks/kernel-matrix-benchmarks/blob/master/plot.py).

On my side, I will try to get the GPU side of things running...

See you soon!
Jean

**Comment by maltezfaria** [comment]
@jeanfeydy: thanks. I will work on the first bench, and will make a PR when that is ready.

**Comment by maltezfaria** [comment]
I fixed the code for generating points on the sphere (see the #4). 

Perhaps we should add an option to the `create_dataset.py` script to generate all datasets at once? Also, I worked on the `README` a bit, mostly cleaning up some things. We have to decide how to "host" the dataset and present it in a table. Unlike the original repo, we will have quite a few datasets... 

I suggest we plan a meeting once you sort out the GPU issues to polish the tests and datasets. Right now there are some strange things happening with the `BruteForce` approach which have to be fixed (e.g. it. gives `nan` on low precision).

**Comment by jeanfeydy** [comment]
Hi @maltezfaria,

Thanks a lot!
To answer your points:

- Please feel free to merge the new sphere generation method - it certainly seems to be more relevant than iid sampling.
- Please feel free to add an option for the `create_dataset.py` script: that’s a very good idea.
- With respect to dataset hosting: I have 100Gb of hosting space on the website `kernel-matrix-benchmarks.com`, so this should be fine I think? (My understanding is that the final benchmark will weigh between 1Gb and 10Gb.)
- I will certainly work on a better BruteForce baseline, implemented with KeOps on the GPU. I have just received my quota increase for GPU instances on AWS, so I should make good progress over the next two weeks.

Best regards,
Jean
